app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  before_save :update_post_counter
+
+  scope :get_5_comments, ->(post) { post.comments.order('created_at desc').limit(5) }
+
+  def update_post_counter
+    poster = User.find_by_id(author.id)
+    poster.posts_count = poster.posts_count.to_i + 1
+    poster.save
+  end
```

Comment:
- [ ] Since we are calling this method in the callback above, we should make it private then